client/src/client.rs

```diff
@@ -536,9 +537,13 @@ where
         Ok(result)
     }
 
-    fn wait_for_peers(&self, count: usize) -> Result<(), handle::Error> {
+    fn wait_for_peers(
+        &self,
+        count: usize,
+        supported_services: Service,
```

Comment:
We call this `required_services` elsewhere.

Reply:
Naming it as `services` overwrites 
```rust
Event::PeerManager(peermgr::Event::PeerNegotiated { addr, **services** })
```
I decided to leave it as it is since wallet is only seeing the trait definitions

Reply:
Oh and I almost forgot: we actually want this to be `Into<ServiceFlags>`, not `Services`, that way you can pass either of the two types!